pkg/controller/pxc/controller.go

```diff
@@ -412,6 +467,45 @@ func (r *ReconcilePerconaXtraDBCluster) deploy(cr *api.PerconaXtraDBCluster) err
 		return fmt.Errorf("get PXC stateful set: %v", err)
 	}
 
+	if (cr.Spec.HAProxy != nil && cr.Spec.HAProxy.Enabled) &&
+		(cr.Spec.ProxySQL != nil && cr.Spec.ProxySQL.Enabled) {
+		return errors.New("Can't Enable Both HAProxy and ProxySQL please only select one of them")
+	}
```

Comment:
this check should be a part of https://github.com/percona/percona-xtradb-cluster-operator/blob/bc4f532d03508dd03165c2b2786b7a22cfd724a4/pkg/apis/pxc/v1/pxc_types.go#L325